### PR TITLE
Add Docs.MS Release Steps

### PR DESCRIPTION
**Pull request by scbedd**
And refactor the artifact staging a bit.

@Azure/azure-sdk-eng 

@weshaggard I've removed a bunch of `New-Item` invocations in favor of a `Copy-Item` with `-Force` parameter. If we know we're copying into a clean directory, do you see any downsides to this approach?

[Example Successful Release](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=396256&view=logs&j=5ccebfdc-da53-51d4-fecb-9480b00e8628&t=63e51a0b-5a43-5cd8-50a8-fa0b432d926a)

### CODE REVIEW

**Comment by weshaggard** [comment]
>@weshaggard I've removed a bunch of New-Item invocations in favor of a Copy-Item with -Force parameter. If we know we're copying into a clean directory, do you see any downsides to this approach?

Does copy-item  -force also create directories?

**Comment by weshaggard** [comment]
>Does copy-item -force also create directories?

Based on a quick test it doesn't appear to so you still need to new-items to create the directories.

**Comment by scbedd** [comment]
@weshaggard I completed some local testing. I'm fairly confident that my approach for `New-Item` will be safe. I've got everything the way I want it except for the publish to nuget (which got commented). I'll uncomment, squash, and checkin when I have the successful release.

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 1]
```diff
@@ -151,14 +157,54 @@ stages:
                   deploy:
                     steps:
                       - checkout: self
+                      - template: /eng/pipelines/templates/steps/stage-artifacts.yml
+                        parameters:
+                          SourceFolder: ${{parameters.ArtifactName}}-signed
+                          TargetFolder: ${{artifact.safeName}}
+                          FileFilter: ${{artifact.name}}.[0-9]*.[0-9]*.[0-9]*
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:164]: Should we make this value the default filter in the template?

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 2]
```diff
@@ -151,14 +157,54 @@ stages:
                   deploy:
                     steps:
                       - checkout: self
+                      - template: /eng/pipelines/templates/steps/stage-artifacts.yml
+                        parameters:
+                          SourceFolder: ${{parameters.ArtifactName}}-signed
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:162]: We should consider making "-signed" as a default in the template so we don't accidently forget to pass it.

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-net-release.yml:162]: Fair point, but it's super minor and I'd rather not retest this again.

File: eng/pipelines/templates/stages/archetype-net-release.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-net-release.yml:164]: We could, but it I'm pretty confident that it'd be a whole nother can of worms to attempt to resolve the inevitable wonky yml rendering issues.

**Review by weshaggard — APPROVED** [review]

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**